packages/trader/src/App/Components/Routes/__tests__/binary-routes.spec.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { render, waitFor, screen } from '@testing-library/react';
+import BinaryRoutes from '../binary-routes';
+import TraderProviders from '../../../../trader-providers';
+import { mockStore } from '@deriv/stores';
+import { Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+
+jest.mock('Modules/Contract', () => {
+    return {
+        __esModule: true,
+        default: () => <div>Contract Details</div>,
+    };
+});
+
+jest.mock('Modules/Trading', () => {
+    return {
+        __esModule: true,
+        default: () => <div>Trader</div>,
+    };
+});
+
+jest.mock('Modules/Page404', () => {
+    return {
+        __esModule: true,
+        default: () => <div>Error 404</div>,
+    };
+});
+
+describe('BinaryRoutes', () => {
+    it('should render contract route', async () => {
+        const history = createMemoryHistory();
+        history.push('/contract/12');
+        render(
+            <TraderProviders store={mockStore({})}>
+                <Router history={history}>
+                    <BinaryRoutes is_logged_in is_logging_in />
+                </Router>
+            </TraderProviders>
+        );
+        await waitFor(() => {
+            expect(screen.getByText('Contract Details')).toBeInTheDocument();
+        });
+    });
+
+    it('should render trade route', async () => {
+        const history = createMemoryHistory();
+        history.push('/');
+        render(
+            <TraderProviders store={mockStore({})}>
+                <Router history={history}>
+                    <BinaryRoutes is_logged_in is_logging_in />
+                </Router>
+            </TraderProviders>
+        );
+        await waitFor(() => {
+            expect(screen.getByText('Trader')).toBeInTheDocument();
+        });
+    });
+
+    it('should render 404 route', async () => {
+        const history = createMemoryHistory();
+        history.push('/non-existent-path');
+        render(
+            <TraderProviders store={mockStore({})}>
+                <Router history={history}>
+                    <BinaryRoutes is_logged_in is_logging_in />
+                </Router>
+            </TraderProviders>
+        );
+        await waitFor(() => {
+            expect(screen.getByText('Error 404')).toBeInTheDocument();
+        });
+    });
+
+    it('should redirect to login if not logged in and not logging in', async () => {
+        const history = createMemoryHistory();
+        history.push('/contract/12');
+        render(
+            <TraderProviders store={mockStore({})}>
+                <Router history={history}>
+                    <BinaryRoutes is_logged_in={false} is_logging_in={false} />
+                </Router>
+            </TraderProviders>
```

Comment:
If `<TraderProviders store={mockStore({})}><Router history={history}> <BinaryRoutes/></Router></Router></TraderProviders>` repeats in each test case, it can be extracted into the separate function. Parameters for <BinaryRoutes> might be passed as arguments.
Please, ignore if not relevant

Reply:
Updated 🙏